From 3f8eeab38172af0e6e9ecab780741814f038d7e7 Mon Sep 17 00:00:00 2001
From: Anish Trivedi <anish@freescale.com>
Date: Fri, 15 Jul 2011 17:41:29 -0500
Subject: [PATCH 229/505] ENGR00153209 Upgrade RTC autorun script for MX6

Changed autorun-rtc.sh script so that the periodic interrupt tests
are not run for MX6 since the SNVS RTC (low power) hw does not support
periodic interrupts. Also, since the time set notification ioctls
are only implemented on MX5 SRTC driver, do not run the tests
associated with these ioctls for other platforms.

Signed-off-by: Anish Trivedi <anish@freescale.com>
---
 test/mxc_rtc/autorun-rtc.sh | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/test/mxc_rtc/autorun-rtc.sh b/test/mxc_rtc/autorun-rtc.sh
index aff1b39..219fc94 100755
--- a/test/mxc_rtc/autorun-rtc.sh
+++ b/test/mxc_rtc/autorun-rtc.sh
@@ -8,11 +8,11 @@ source /unit_tests/test-utils.sh
 STATUS=0
 
 echo "$(platform)"
-if [ "$(platform)" == IMX37_3STACK ] || [ "$(platform)" == IMX51 ] \
-    || [ "$(platform)" == IMX53 ] || [ "$(platform)" == IMX50 ]; then
+if [ "$(platform)" != IMX31ADS ] && [ "$(platform)" != IMX32ADS ] \
+    && [ "$(platform)" != IMX25_3STACK ] && [ "$(platform)" != IMX35_3STACK ]; then
 	rtc_test_param=--no-periodic
 	# For kernel version 2.6.38 and higher, number of interrupts expected will be 11
-	VER=`echo $(kernel_version) | cut -d. -f3`
+	VER=`echo $(kernel_version) | cut -d. -f3 | cut -d- -f1`
 	REF_VER=38
 	if [ $VER -ge $REF_VER ]; then
 		RTC_IRQS_EXPECTED=11
@@ -47,7 +47,9 @@ if [ "$RTC_IRQS" != "$RTC_IRQS_EXPECTED" ]; then
 else
 	echo "checking rtc interrupts PASS"
        #RTC wait for time set notification test
-        run_testcase "./rtc_timesetnotification_test.out"
+	if [ "$(platform)" == IMX50 ] || [ "$(platform)" == IMX51 ] || [ "$(platform)" == IMX53 ]; then
+          run_testcase "./rtc_timesetnotification_test.out"
+      fi
 fi
 
 print_status
-- 
1.8.0

